### PR TITLE
CAB-3048 implement v3/delete endpoint

### PR DESCRIPTION
**Pull request by maximede**

### CODE REVIEW

File: src/main/java/edu/uw/edm/contentapi2/controller/content/v3/ItemV3Controller.java [hunk 1]
```diff
@@ -51,6 +57,25 @@ public DocumentSearchResults _searchItems(
         return documentSearchResults;
     }
 
+    @RequestMapping(value = "/{itemId}", method = RequestMethod.DELETE)
+    public ResponseEntity expireItem(
```
Comment (agagne) [inline src/main/java/edu/uw/edm/contentapi2/controller/content/v3/ItemV3Controller.java]:
```suggestion
    public ResponseEntity deleteItem(
```

File: src/main/java/edu/uw/edm/contentapi2/controller/content/v3/ItemV3Controller.java [hunk 2]
```diff
@@ -51,6 +57,25 @@ public DocumentSearchResults _searchItems(
...
+            @PathVariable("itemId") String itemId,
+            @RequestParam(value = "immediate", required = false, defaultValue = "false") Boolean immediate,
+            @AuthenticationPrincipal User user) throws RepositoryException {
+
+
+        if (immediate != null) {
```
Comment (agagne) [inline src/main/java/edu/uw/edm/contentapi2/controller/content/v3/ItemV3Controller.java]: is this possible? if you're defaultValue is "false"?

Comment (maximede) [reply src/main/java/edu/uw/edm/contentapi2/controller/content/v3/ItemV3Controller.java]: duh...

File: src/main/java/edu/uw/edm/contentapi2/controller/content/v3/ItemV3Controller.java [hunk 3]
```diff
@@ -51,6 +57,25 @@ public DocumentSearchResults _searchItems(
         return documentSearchResults;
     }
 
+    @RequestMapping(value = "/{itemId}", method = RequestMethod.DELETE)
+    public ResponseEntity expireItem(
+            @PathVariable("itemId") String itemId,
+            @RequestParam(value = "immediate", required = false, defaultValue = "false") Boolean immediate,
```
Comment (agagne) [inline src/main/java/edu/uw/edm/contentapi2/controller/content/v3/ItemV3Controller.java]: deprecate immediate?

**Pull request merged**